app/src/main/java/com/babylon/wallet/android/presentation/wallet/WalletViewModel.kt

```diff
@@ -258,88 +279,94 @@ class WalletViewModel @Inject constructor(
             walletUiState.copy(isFiatBalancesEnabled = isEnabled)
         }
     }
-}
 
-internal sealed interface WalletEvent : OneOffEvent {
-    data class NavigateToMnemonicBackup(val factorSourceId: FactorSourceId.Hash) : WalletEvent
-    data class NavigateToMnemonicRestore(val factorSourceId: FactorSourceId.Hash) : WalletEvent
+    private fun loadingResources(isRefreshing: Boolean): WalletUiState = state.value.copy(
+        accountUiItems = buildAccountUiItems(),
+        isLoading = accountsWithAssets == null,
+        isRefreshing = isRefreshing
+    )
 
-    data object ShowNpsSurvey : WalletEvent
-}
+    private fun onResourcesReceived(): WalletUiState {
+        val accountUiItems = buildAccountUiItems()
 
-data class WalletUiState(
-    private val accountsWithAssets: List<AccountWithAssets>? = null,
-    private val accountsAddressesWithAssetsPrices: Map<AccountAddress, List<AssetPrice>?>? = null,
-    private val loading: Boolean = true,
-    private val refreshing: Boolean = false,
-    private val entitiesWithSecurityPrompt: List<EntityWithSecurityPrompt> = emptyList(),
-    private val factorSources: List<com.radixdlt.sargon.FactorSource> = emptyList(),
-    val isRadixBannerVisible: Boolean = false,
-    val isBackupWarningVisible: Boolean = false,
-    val isFiatBalancesEnabled: Boolean = true,
-    val uiMessage: UiMessage? = null,
-    val isNpsSurveyShown: Boolean = false
-) : UiState {
+        return state.value.copy(
+            isLoading = false,
+            isRefreshing = false,
+            accountUiItems = accountUiItems,
+            totalFiatValueOfWallet = buildTotalFiatValue()
+        )
+    }
 
-    val accountsAndAssets: List<AccountWithAssets>
-        get() = accountsWithAssets.orEmpty()
+    private fun onResourcesError(error: Throwable?): WalletUiState = state.value.copy(
+        uiMessage = UiMessage.ErrorMessage(error),
+        accountUiItems = state.value.accountUiItems.map { account ->
+            if (account.assets == null) {
+                // If assets don't exist leave them empty
+                account.copy(assets = Assets())
+            } else {
+                // Else continue with what the user used to see before the refresh
+                account
+            }
+        },
+        isLoading = false,
+        isRefreshing = false
+    )
```

Comment:
Could you rename these methods to replace Resources with Assets? It is a remnant of the past

Reply:
Good point! replaced